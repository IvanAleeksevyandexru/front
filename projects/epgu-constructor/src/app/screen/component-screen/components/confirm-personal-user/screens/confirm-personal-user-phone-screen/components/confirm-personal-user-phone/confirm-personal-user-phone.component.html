<div class="flex-row">
  <epgu-constructor-label for="phone">{{ label }}</epgu-constructor-label>
  <a href="javascript:void(0);"
     class="button-change"
     (click)="handleClick()"
     *ngIf="isEditButtonShown">
    Редактировать
  </a>
</div>

<lib-base-masked-input
  class="content content__input"
  [disabled]="true"
  [(ngModel)]="data"
  [mask]="phoneMask"
></lib-base-masked-input>



