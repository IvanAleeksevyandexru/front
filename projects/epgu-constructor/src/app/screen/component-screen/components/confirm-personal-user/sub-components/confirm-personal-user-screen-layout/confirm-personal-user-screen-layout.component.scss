@import "src/styles/core";


$button-mt: 5px;

:host {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
}

:host::ng-deep {
  $actions-mx: 10px;
  .footer {
    display: flex;
    justify-content: center;
    flex-direction: column;
    flex-wrap: wrap;
    margin: auto $actions-mx 0;
    @include tablets-up {
      flex-direction: row;
      margin-top: 16px;
    }

    &:empty {
      display: none;
    }

    .confirm-personal-user-button {
      @include tablets-up {
        margin: 5px $actions-mx;
      }
    }
  }
}
