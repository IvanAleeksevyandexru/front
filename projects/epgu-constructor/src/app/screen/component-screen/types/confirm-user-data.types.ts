import { ComponentItem } from '../../../services/api/form-player-api/form-player-api.types';

export interface ConfirmUserData extends ComponentItem{
  attrs: ConfirmUserDataAttrs;
}
export interface ConfirmUserDataAttrs {
  fields: Array<ConfirmUserDataField>;
  actions: Array<ConfirmUserDataActions>;
}

export interface ConfirmUserDataActions {
  label: string;
  method: string;
}

export interface ConfirmUserDataField {
  fieldName: string;
  label: string;
}

export interface ConfirmUserDataAdaptiveField {
  label: string;
  value: string;
}

export interface ConfirmUserDataFieldsState {
  groupName: string;
  list: Array<ConfirmUserDataAdaptiveField>;
}
