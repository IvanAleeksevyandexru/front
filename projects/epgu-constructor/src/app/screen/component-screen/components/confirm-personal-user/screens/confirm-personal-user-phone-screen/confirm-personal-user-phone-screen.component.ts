import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { ComponentItem } from '../../../../../../services/api/form-player-api/form-player-api.types';
import { ComponentStateService } from '../../../../../../services/component-state/component-state.service';

@Component({
  selector: 'epgu-constructor-confirm-personal-user-phone-screen',
  templateUrl: './confirm-personal-user-phone-screen.component.html',
  styleUrls: ['./confirm-personal-user-phone-screen.component.scss'],
})
export class ConfirmPersonalUserPhoneScreenComponent implements OnInit {
  @Input() data: ComponentItem;
  @Input() errors: object;
  @Output() nextStepEvent = new EventEmitter();

  constructor(private componentStateService: ComponentStateService) {}

  ngOnInit(): void {
    // TODO возможно стоит переместить во внутриь компонента
    this.componentStateService.state = this.data.value;
  }

  clickToAction(action: NotificationAction) {
    console.log(action);
  }
}
