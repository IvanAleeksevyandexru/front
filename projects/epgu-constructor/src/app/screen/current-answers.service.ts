import { Injectable } from '@angular/core';
import { ComponentBase } from './screen.types';
import { SlotInterface } from '../component/unique-screen/components/time-slots/time-slots.types';

@Injectable()
export class CurrentAnswersService {

  // <-- component
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  private _state: any;
  private _isValid: boolean;

  get state () {
    return this._state;
  }

  // TODO привести все к json строке
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  set state (val: any) {
    this._state = val;
  }

  get isValid (): boolean {
    return this._isValid;
  }
  set isValid (val: boolean) {
    this._isValid = val;
  }

  constructor() { }
}
