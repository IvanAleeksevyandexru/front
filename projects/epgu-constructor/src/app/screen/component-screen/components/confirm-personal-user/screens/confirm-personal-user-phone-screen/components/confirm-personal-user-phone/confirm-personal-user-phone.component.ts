import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
import { ComponentStateService } from '../../../../../../../../services/component-state/component-state.service';
import { NavigationService } from '../../../../../../../../shared/services/navigation/navigation.service';
import { ScreenService } from '../../../../../../../screen.service';
import { Navigation } from '../../../../../../../../form-player.types';

// TODO удалить этот компонент в пользу CUSTOM SCREEN c поддержкой actions;
@Component({
  selector: 'epgu-constructor-confirm-personal-user-phone',
  templateUrl: './confirm-personal-user-phone.component.html',
  styleUrls: ['./confirm-personal-user-phone.component.scss'],
})
export class ConfirmPersonalUserPhoneComponent implements OnChanges {
  @Input() label: string;
  @Input() data: string;
  @Input() actions: Array<{ label: string; value: string; action: string }>; // TODO HARDCODE
  @Input() isEditButtonShown: boolean;

  phoneMask = [
    '+',
    '7',
    ' ',
    '(',
    /[1-9]/,
    /\d/,
    /\d/,
    ')',
    ' ',
    /\d/,
    /\d/,
    /\d/,
    '-',
    /\d/,
    /\d/,
    '-',
    /\d/,
    /\d/,
  ];

  constructor(
    private componentStateService: ComponentStateService,
    private screenService: ScreenService,
    private navigationService: NavigationService,
  ) {}

  handleClick() {
    const action = this.actions[0];
    const navigation: Navigation = {
      payload: this.getComponentStateForNavigate(),
      options: { url: action.action },
    };
    this.navigationService.nextStep.next(navigation);
  }

  ngOnChanges(changes: SimpleChanges): void {
    if (changes.data?.currentValue) {
      this.componentStateService.state = this.data;
    }
  }

  getComponentStateForNavigate() {
    return {
      [this.screenService.component.id]: {
        visited: true,
        value: this.componentStateService.state,
      },
    };
  }
}
