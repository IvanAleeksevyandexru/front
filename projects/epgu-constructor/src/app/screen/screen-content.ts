import { ComponentDto, DisplayDto } from '../services/api/form-player-api/form-player-api.types';
import { ScreenStore, ScreenTypes } from './screen.types';
import { BehaviorSubject } from 'rxjs';

export class ScreenContent {

  private _display = new BehaviorSubject<DisplayDto>(null);
  public get display() {
    return this._display.getValue();
  }
  public set display(val: DisplayDto) {
    this._display.next(val);
  }
  private display$ = this._display.asObservable();

  private _header = new BehaviorSubject<string>(null);
  public get header() {
    return this._header.getValue();
  }
  public set header(val: string) {
    this._header.next(val);
  }
  public header$ = this._header.asObservable();

  private _submitLabel = new BehaviorSubject<string>(null);
  public get submitLabel() {
    return this._submitLabel.getValue();
  }
  public set submitLabel(val: string) {
    this._submitLabel.next(val);
  }
  public submitLabel$ = this._submitLabel.asObservable();

  private _screenType = new BehaviorSubject<ScreenTypes>(null);
  public get screenType() {
    return this._screenType.getValue();
  }
  public set screenType(val: ScreenTypes) {
    this._screenType.next(val);
  }
  public screenType$ = this._screenType.asObservable();

  private _orderId = new BehaviorSubject<string>(null);
  public get orderId() {
    return this._orderId.getValue();
  }
  public set orderId(val: string) {
    this._orderId.next(val);
  }
  public orderId$ = this._orderId.asObservable();

  private _component = new BehaviorSubject<ComponentDto>(null);
  public get component() {
    return this._component.getValue();
  }
  public set component(val: ComponentDto) {
    this._component.next(val);
  }
  public component$ = this._component.asObservable();

  private _componentType = new BehaviorSubject<string>(null);
  public get componentType() {
    return this._componentType.getValue();
  }
  public set componentType(val: string) {
    this._componentType.next(val);
  }
  public componentType$ = this._componentType.asObservable();

  constructor() {}

  updateScreenContent(screenStore: ScreenStore) {
    const { display = {} as any, orderId } = screenStore;
    const { header, submitLabel, type, components = [] } = display;
    this.display = display;
    this.header = header;
    this.submitLabel = submitLabel;
    this.screenType = type;
    this.component = components[0];
    this.componentType = components[0]?.type;
    this.orderId = orderId;
  }
}
