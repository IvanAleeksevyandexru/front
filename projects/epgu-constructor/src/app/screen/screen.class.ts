import { Injector } from '@angular/core';
import { NavigationService } from '../core/services/navigation/navigation.service';
import { UnsubscribeService } from '../core/services/unsubscribe/unsubscribe.service';
import { takeUntil } from 'rxjs/operators';
import { ScreenService } from './screen.service';
import { NavigationPayload } from '../form-player/form-player.types';

// Желательно больше не наследоваться потому что это может усложнить логику.
export abstract class ScreenClass {
  public navigationService: NavigationService;
  public ngUnsubscribe$: UnsubscribeService;
  public screenService: ScreenService;

  protected constructor(public injector: Injector) {
    this.navigationService = this.injector.get(NavigationService);
    this.ngUnsubscribe$ = this.injector.get(UnsubscribeService);
    this.screenService = this.injector.get(ScreenService);
  }

  subscribeToNavigatePrev() {
    this.navigationService.clickToBack$
      .pipe(takeUntil(this.ngUnsubscribe$))
      .subscribe(() => this.prevStep());
  }

  /**
   * Возвращение на экран назад
   */
  prevStep(): void {
    this.navigationService.prevStep.next();
  }

  abstract nextStep(data?: NavigationPayload): void;
}
