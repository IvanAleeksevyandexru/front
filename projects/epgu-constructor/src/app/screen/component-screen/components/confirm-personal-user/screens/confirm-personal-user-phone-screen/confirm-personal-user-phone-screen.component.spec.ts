import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { MockComponent } from 'ng-mocks';
import { ConfirmPersonalUserPhoneScreenComponent } from './confirm-personal-user-phone-screen.component';
import { ComponentStateService } from '../../../../../component-state.service';
// eslint-disable-next-line max-len
import { ConfirmPersonalUserScreenLayoutComponent } from '../../sub-components/confirm-personal-user-screen-layout/confirm-personal-user-screen-layout.component';
import { ConfirmPersonalUserPhoneComponent } from './components/confirm-personal-user-phone/confirm-personal-user-phone.component';
import { ComponentBase } from '../../../../../screen.types';


describe('ConfirmPersonalUserPhoneScreenComponent', () => {
  let component: ConfirmPersonalUserPhoneScreenComponent;
  let fixture: ComponentFixture<ConfirmPersonalUserPhoneScreenComponent>;
  const mockData: ComponentBase = {
    attrs: {},
    id: '',
    label: '',
    type: '',
    value: ''
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [ CUSTOM_ELEMENTS_SCHEMA ], // TODO: remove this line when resolve issue with @ifc/plugin and @ifc/common dependencies
      declarations: [
        ConfirmPersonalUserPhoneScreenComponent,
        ConfirmPersonalUserScreenLayoutComponent
      ],
      providers: [ ComponentStateService ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ConfirmPersonalUserPhoneScreenComponent);
    component = fixture.componentInstance;
    component.data = mockData;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
