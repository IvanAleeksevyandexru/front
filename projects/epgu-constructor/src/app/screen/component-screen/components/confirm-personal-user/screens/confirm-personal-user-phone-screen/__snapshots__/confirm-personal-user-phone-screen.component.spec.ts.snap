// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPersonalUserPhoneScreenComponent check snapshot 1`] = `
<epgu-constructor-confirm-personal-user-phone-screen
  componentStateService={[Function ComponentStateService]}
  data={[Function Object]}
  nextStepEvent={[Function EventEmitter_]}
>
  <epgu-constructor-confirm-personal-user-screen-layout>
    <div
      class="body"
    >
      <epgu-constructor-confirm-personal-user-phone />
    </div>
    <div
      class="footer"
    >
      
    </div>
  </epgu-constructor-confirm-personal-user-screen-layout>
</epgu-constructor-confirm-personal-user-phone-screen>
`;
